src/v2/Apps/Unsubscribe/UnsubscribeApp.tsx

```diff
@@ -7,37 +7,37 @@ import { UnsubscribeApp_me } from "v2/__generated__/UnsubscribeApp_me.graphql"
 import { UnsubscribeLoggedInFragmentContainer } from "./Components/UnsubscribeLoggedIn"
 import { UnsubscribeLoggedOut } from "./Components/UnsubscribeLoggedOut"
 
+const UnsubscribeFallback = () => {
+  return (
+    <Message variant="error" my={4}>
+      Please sign in to update your email preferences
+    </Message>
+  )
+}
+
 interface UnsubscribeAppProps {
   me: UnsubscribeApp_me | null
 }
 
 export const UnsubscribeApp: React.FC<UnsubscribeAppProps> = ({ me }) => {
   const { match } = useRouter()
   const { authentication_token: authenticationToken } = match.location.query
-
-  if (!me && !authenticationToken) {
-    return (
-      <Message variant="error" my={4}>
-        Unable to update your email preferences
-      </Message>
-    )
-  }
+  const showFallback = !me && !authenticationToken
+  const showLoggedIn = me && !authenticationToken
+  const showLoggedOut = !me && authenticationToken
 
   return (
     <>
+      {showFallback && <UnsubscribeFallback />}
```

Comment:
In this case it doesn't matter much, but in general, rather than redeclaring a component in a variable in the function body, just inline it (or extract it, in this case it's simple enough though). (This _can_ create issues when images are used)

Reply:
Hrm, not sure I follow this feedback - maybe some examples would help?

Reply:
```suggestion
      {showFallback && (
        <Message variant="error" my={4}>
          Please sign in to update your email preferences
        </Message>
      )}
```

Reply:
But you'd have to update the specs, blah blah. Kind of just a general comment: I see this a lot and we should avoid doing it because it can cause very mysterious issues with re-renders. In this case; it's not really a big deal.

Reply:
> rather than redeclaring a component in a variable in the function body

I don't think I did this - I think I extracted it to a variable in the file, no? Or are you saying I should have extracted it all the way into a separate file?

Reply:
Oh, lol you're absolutely right. __This is fine ignore me__

Reply:
As a related aside, I was debugging some serious sidebar weirdness in the Palette docs and lo and behold, declared the component in the function, versus outside of it.